### ISSUE TITLE
Transaction templates

### ISSUE THREAD

**Issue by frandavid100** [mvarnagiris/financius#6]
Predefined quickly accessible templates for transactions

Original:

In order to make creating transactions as quick and easy as possible, the editor should autocomplete certain fields based on the data it already has. Here are some mockups showing what I mean:

![financius editor 7](https://f.cloud.github.com/assets/694794/1509643/92cb5840-4a57-11e3-969a-f2d18e7b3e86.png)

First, when you start filling the Concept field, a dropdown should offer you to autocomplete from your groups and from the concept field of previous transactions.

If you choose a group, the "To" field should autofill (as seen in issue #5) with the group's category and participants.

If you choose a concept, all the fields should autofill with that concept's previous values.

![financius editor 8](https://f.cloud.github.com/assets/694794/1509656/4871a35c-4a58-11e3-92fa-da5e56066019.png)

Second, when you start filling the From field, you should be able to autocomplete from your income categories, accounts and contacts.

![financius editor 9](https://f.cloud.github.com/assets/694794/1509663/79618680-4a58-11e3-9c01-8237c7b12b7d.png)

Third, when you start filling the To field, you should be able to autocomplete from your expense categories, accounts and contacts.

**Comment by ansariNo1** [comment]
What I believe would be a great enhancement for the Transaction screen is to make the app auto predict the category/subcategory based on what I write in the Notes field, depending on history data of course. This way I don't have to select the category everytime unless I don't have something to write in the Notes.

**Comment by frandavid100** [comment]
@ansariNo1 note that, for that to work, the Notes field would have to be moved to the top in order to be filled before the other ones.

**Comment by ansariNo1** [comment]
Exactly, I think It's better this way. Since some people may not want to write any notes, the field doesn't have to be permanent. It can be a [+] button that expands to a text field if you click it.

**Comment by leoetlino** [comment]
**:+1: This would really make Financius the best app!**

It has everything it takes: Holo style, no ads, loads of features, open-source app, categories, etc - it's only missing this feature!

**Comment by mvarnagiris** [comment]
I'm demoting this feature to suggestion for now, as I'm not sure yet how to incorporate that in the app

**Comment by shashwatblack** [comment]
This feature should make everything very convenient.

You first define a template which is nothing but the all other fields (category, income/expense, tags, From, To, Note etc) pre-saved. 
When creating a new transaction, there's an inconspicuous toggle at the top (now showing Default Templates - all blanks) which allows you to select one from your saved templates. This would just fill the fields in the form.
